source/distributed/tria.cc

```diff
@@ -4532,6 +4650,50 @@ namespace parallel
     }
 
     template <int dim, int spacedim>
+    void
+    Triangulation<dim,spacedim>::
+    get_cell_weights(std::vector<unsigned int> &weights)
+    {
+      Assert(weights.size() == 0,
+             ExcMessage("You are trying to calculate weights, but the given vector is not empty. Refusing to overwrite content."));
+      weights.reserve(this->n_active_cells());
```

Comment:
ah, nice, I had already wondered whether you'd do this `reserve` :-)

I think you don't actually need `n_active_cells()` slots, though. Aren't you only going to push `n_locally_owned_active_cells()` into the vector?